### ISSUE TITLE
Auto-versioning version source based on links

### ISSUE THREAD

**Issue by dcoraboeuf** [nemerosa/ontrack#1339]
Create a `versionSource` implementation able to get the version from a link of the build being promoted.

For example:

```yaml
versionSource: link/my-dependency
```

In this case, we can get the version from the promoted build by getting the version of the link to the `my-dependency` project. If this link does not exist, we abort the auto-versioning.

The linked build can itself use a custom version source:

```yaml
versionSource: link/my-dependency->metaInfo/rpmVersion
```

In this case, once we have the dependency, we want its `rpmVersion` meta information.

We can even chain the dependencies:

```yaml
versionSource: link/my-dependency->link/other-dependency->metaInfo/rpmVersion
```

The general format for this version source is:

```
link/$project[/$qualifier][->$source]
```

**Comment by dcoraboeuf** [comment]
Candidate regex is:

```
^link\/([a-zA-Z0-9_\-]+)(?:\/([a-zA-Z0-9_\-]+))?(?:->(.*))?$
```

**Comment by dcoraboeuf** [comment]
Available in 4.9.9
